File: CS/S170863/Form1.cs
using DevExpress.Utils;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using System;
using System.Data;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            vGridControl1.DataSource = CreateTable(6);
            RepositoryItemCheckEdit riCheckEdit = new RepositoryItemCheckEdit();
            new RowInplaceEditorHelper(vGridControl1.GetRowByFieldName("Check"), riCheckEdit);
            riCheckEdit.AllowFocused = false;
            riCheckEdit.EditValueChanged += riCheckEdit_EditValueChanged;
            RepositoryItemColorEdit riColorEdit = new RepositoryItemColorEdit();
            new RowInplaceEditorHelper(vGridControl1.GetRowByFieldName("Name"), riColorEdit);
            riColorEdit.ColorAlignment = HorzAlignment.Center;
            riColorEdit.EditValueChanged += riColorEdit_EditValueChanged;
        }

        void riColorEdit_EditValueChanged(object sender, EventArgs e) {
                vGridControl1.GetRowByFieldName("Name").Appearance.BackColor2 = (sender as ColorEdit).Color;
        }

        void riCheckEdit_EditValueChanged(object sender, EventArgs e) {
            for(int i = 0; i < vGridControl1.RecordCount; i++) 
                vGridControl1.SetCellValue(vGridControl1.GetRowByFieldName("Check"), i, (sender as CheckEdit).Checked);            
        }

        private DataTable CreateTable(int recordCount) {
            DataTable dataTable = new DataTable();
            dataTable.Columns.Add("Name", typeof(string));
            dataTable.Columns.Add("ID", typeof(int));
            dataTable.Columns.Add("Check", typeof(bool));
            dataTable.Columns.Add("Date", typeof(DateTime));
            for(int i = 0; i < recordCount; i++)
                dataTable.Rows.Add(new object[] { String.Format("Name{0}", i), i, (i % 2 == 1 ? true : false), DateTime.Now.AddDays(i) });
            return dataTable;
        }
    }
}

File: VB/S170863/Form1.vb
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository

Namespace WindowsFormsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            vGridControl1.DataSource = CreateTable(6)
            Dim riCheckEdit As New RepositoryItemCheckEdit()
            Dim TempRowInplaceEditorHelper1 As RowInplaceEditorHelper = New RowInplaceEditorHelper(vGridControl1.GetRowByFieldName("Check"), riCheckEdit)
            riCheckEdit.AllowFocused = False
            AddHandler riCheckEdit.EditValueChanged, AddressOf riCheckEdit_EditValueChanged
            Dim riColorEdit As New RepositoryItemColorEdit()
            Dim TempRowInplaceEditorHelper2 As RowInplaceEditorHelper = New RowInplaceEditorHelper(vGridControl1.GetRowByFieldName("Name"), riColorEdit)
            riColorEdit.ColorAlignment = HorzAlignment.Center
            AddHandler riColorEdit.EditValueChanged, AddressOf riColorEdit_EditValueChanged
        End Sub

        Private Sub riColorEdit_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
                vGridControl1.GetRowByFieldName("Name").Appearance.BackColor2 = (TryCast(sender, ColorEdit)).Color
        End Sub

        Private Sub riCheckEdit_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            For i As Integer = 0 To vGridControl1.RecordCount - 1
                vGridControl1.SetCellValue(vGridControl1.GetRowByFieldName("Check"), i, (TryCast(sender, CheckEdit)).Checked)
            Next i
        End Sub

        Private Function CreateTable(ByVal recordCount As Integer) As DataTable
            Dim dataTable As New DataTable()
            dataTable.Columns.Add("Name", GetType(String))
            dataTable.Columns.Add("ID", GetType(Integer))
            dataTable.Columns.Add("Check", GetType(Boolean))
            dataTable.Columns.Add("Date", GetType(Date))
            For i As Integer = 0 To recordCount - 1
                dataTable.Rows.Add(New Object() { String.Format("Name{0}", i), i, (If(i Mod 2 = 1, True, False)), Date.Now.AddDays(i) })
            Next i
            Return dataTable
        End Function
    End Class
End Namespace